### PR TITLE
openapi-python-client: init at 0.21.5

### PR DESCRIPTION
**Pull request by konradmalik**
## Description of changes

A python application to generate modern Python clients from OpenAPI 3.0 and 3.1 documents.

Homepage: https://github.com/openapi-generators/openapi-python-client

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/op/openapi-python-client/package.nix [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    inherit version;
+    owner = "openapi-generators";
+    repo = "openapi-python-client";
+    rev = "v${version}";
+    hash = "sha256-aYwwj96sHfYf3u5Rpcq7QPZA7O4Kt9vVglV38glxuZ0=";
+
+  };
```
Comment (GaetanLepage) [inline pkgs/by-name/op/openapi-python-client/package.nix:25]:
```suggestion
    hash = "sha256-aYwwj96sHfYf3u5Rpcq7QPZA7O4Kt9vVglV38glxuZ0=";
  };
```

File: pkgs/by-name/op/openapi-python-client/package.nix [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+python3Packages.buildPythonApplication rec {
+  pname = "openapi-python-client";
+  version = "0.21.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    inherit version;
+    owner = "openapi-generators";
+    repo = "openapi-python-client";
+    rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/op/openapi-python-client/package.nix:22]:
```suggestion
    rev = "refs/tags/v${version}";
```

Avoids tag/branch mismatch.

File: pkgs/by-name/op/openapi-python-client/package.nix [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      ruamel-yaml
+      shellingham
+      typer
+      typing-extensions
+    ])
+    ++ [ ruff ];
+
+  build-system = with python3Packages; [
+    hatchling
+  ];
```
Comment (GaetanLepage) [inline pkgs/by-name/op/openapi-python-client/package.nix:46]: According to me, the order should be:
- `nativeBuildInputs`
- `build-system`
- `dependencies`
- `pythonRemoveDeps`

I might not be 100% right though.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Looks pretty good already. Only minor changes to be made.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

- Command: `nixpkgs-review pr 337741`
- Systems: `x86_64-linux`,`aarch64-linux`,`x86_64-darwin`,`aarch64-darwin`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>openapi-python-client</li>
    <li>openapi-python-client.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>openapi-python-client</li>
    <li>openapi-python-client.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>openapi-python-client</li>
    <li>openapi-python-client.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>openapi-python-client</li>
    <li>openapi-python-client.dist</li>
  </ul>
</details>

**Comment by konradmalik** [comment]
offtopic: @GaetanLepage mind sharing how did you generate such a nice `nixpkgs-review` report for all architectures? It's extremely nice

File: pkgs/by-name/op/openapi-python-client/package.nix (hunk 1 above)
Comment (konradmalik) [reply pkgs/by-name/op/openapi-python-client/package.nix]: Thanks for review, and sorry for pinging you directly but this PR lingered for a bit too long 😅 

Changed this and updated to 0.21.5

File: pkgs/by-name/op/openapi-python-client/package.nix (hunk 2 above)
Comment (konradmalik) [reply pkgs/by-name/op/openapi-python-client/package.nix]: Changed this and updated to 0.21.5

File: pkgs/by-name/op/openapi-python-client/package.nix (hunk 3 above)
Comment (konradmalik) [reply pkgs/by-name/op/openapi-python-client/package.nix]: cool, this makes sense, changed this and updated to 0.21.5

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by GaetanLepage** [comment]
> offtopic: @GaetanLepage mind sharing how did you generate such a nice `nixpkgs-review` report for all architectures? It's extremely nice

This is an addition currently under review: https://github.com/Mic92/nixpkgs-review/pull/419

**Pull request merged**